pandas/core/tools/datetimes.py

```diff
@@ -402,46 +405,50 @@ def _convert_listlike(arg, box, format, name=None, tz=tz):
                 # shortcut formatting here
                 if format == '%Y%m%d':
                     try:
-                        result = _attempt_YYYYMMDD(arg, errors=errors)
+                        result = _attempt_YYYYMMDD(obj_arg, errors=errors)
                     except:
                         raise ValueError("cannot convert the input to "
                                          "'%Y%m%d' date format")
 
                 # fallback
                 if result is None:
                     try:
-                        result = tslib.array_strptime(arg, format, exact=exact,
+                        result = tslib.array_strptime(obj_arg, format,
+                                                      exact=exact,
                                                       errors=errors)
                     except tslib.OutOfBoundsDatetime:
                         if errors == 'raise':
                             raise
-                        result = arg
+                        result = obj_arg
                     except ValueError:
                         # if format was inferred, try falling back
                         # to array_to_datetime - terminate here
                         # for specified formats
                         if not infer_datetime_format:
                             if errors == 'raise':
                                 raise
-                            result = arg
+                            result = obj_arg
 
             if result is None and (format is None or infer_datetime_format):
                 result = tslib.array_to_datetime(
-                    arg,
+                    obj_arg,
                     errors=errors,
                     utc=utc,
                     dayfirst=dayfirst,
                     yearfirst=yearfirst,
                     require_iso8601=require_iso8601
                 )
-
+            from pandas import Series
             if is_datetime64_dtype(result) and box:
                 result = DatetimeIndex(result, tz=tz, name=name)
+            # GH 6415
```

Comment:
better to make a wrapper function which handles the utc argument (IOW if its a DTI / Series and utc is true it will localize, otherwise it will pass thru).